File: contactswiththread/src/main/java/com/example/contactswiththread/Data/ContactDAO.java
package com.example.contactswiththread.Data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.contactswiththread.Contact;

import java.util.List;

@Dao
public interface ContactDAO {

    @Insert
    long addContact(Contact contact);

    @Update
    void updateContact(Contact contact);

    @Delete
    void deleteContact(Contact contact);

    @Query("select * from contacts")
    List<Contact> getAllContacts();

    @Query("select * from contacts where contact_id==:contactId ")
    Contact getContact(long contactId);
}

File: contactsinroomlibrary/src/main/java/Data/ContactDAO.kt
package Data

import androidx.room.*
import com.example.contactsinroomlibrary.Contact

@Dao
interface ContactDAO {
    @Insert
    fun addContact(contact: Contact): Long

    @Update
    fun updateContact(contact: Contact)

    @Delete
    fun deleteContact(contact: Contact)

    @get:Query("select * from contacts")
    val allContacts: List<Contact?>

    @Query("select * from contacts where contact_id==:contactId ")
    fun getContact(contactId: Long): Contact
}